jenkins/master/configuration/init.groovy.d/ods-jenkins-shared-library.groovy

```diff
@@ -0,0 +1,57 @@
+#!groovy
+
+// imports
+import hudson.scm.SCM
+import jenkins.model.Jenkins
+import jenkins.plugins.git.GitSCMSource
+import org.jenkinsci.plugins.workflow.libs.*
+import org.jenkinsci.plugins.workflow.libs.LibraryConfiguration
+import org.jenkinsci.plugins.workflow.libs.SCMSourceRetriever
+
+def namespace = "cat /var/run/secrets/kubernetes.io/serviceaccount/namespace".execute().text.trim()
+println "INFO: Jenkins adding build shared lib: ${namespace}"
+
+def buildSharedLibName = "ods-jenkins-shared-library"
+
+def env = System.getenv()
+def buildSharedLibraryRepository = env['SHARED_LIBRARY_REPOSITORY']
+
+def credentialsId = namespace + "-cd-user-with-password"
+// parameters
+def globalLibrariesParameters = [
+  branch:               "production",
+  credentialId:         credentialsId,
+  implicit:             false,
+  name:                 buildSharedLibName,
+  repository:           buildSharedLibraryRepository
+]
+
+// define global library
+GitSCMSource gitSCMSource = new GitSCMSource(
+  globalLibrariesParameters.name,
+  globalLibrariesParameters.repository,
+  globalLibrariesParameters.credentialId,
+  "*",
+  "",
+  false
+)
+
+// define retriever
+SCMSourceRetriever sCMSourceRetriever = new SCMSourceRetriever(gitSCMSource)
+
+// get Jenkins instance
+Jenkins jenkins = Jenkins.getInstance()
+
+// get Jenkins Global Libraries
+def globalLibraries = jenkins.getDescriptor("org.jenkinsci.plugins.workflow.libs.GlobalLibraries")
+
+// define new library configuration
+LibraryConfiguration libraryConfiguration = new LibraryConfiguration(globalLibrariesParameters.name, sCMSourceRetriever)
+libraryConfiguration.setDefaultVersion(globalLibrariesParameters.branch)
+libraryConfiguration.setImplicit(globalLibrariesParameters.implicit)
+
+// set new Jenkins Global Library
+globalLibraries.get().setLibraries([libraryConfiguration])

```

Comment:
In my opinion this statement replaces all existing shared library definitions. I have two cases in mind:
1. Jenkins is restarted.
2. A second library is added by another script (see below).

Reply:
Yup

Reply:
fixed - please check